465 price full sync

bricht opened the pull request:
<!---
    Thank you for contributing to Magento.
    To help us process this pull request we recommend that you add the following information:
     - Summary of the pull request,
     - Issue(s) related to the changes made,
     - Manual testing scenarios
    Fields marked with (*) are required. Please don't remove the template.
-->

<!--- Please provide a general summary of the Pull Request in the Title above -->

### Description (*)
<!---
    Please provide a description of the changes proposed in the pull request.
    Letting us know what has changed and why it needed changing will help us validate this pull request.
-->

### Related Pull Requests
Extends and depends on the following pull request:
https://github.com/magento/commerce-data-export/pull/20

<!-- related pull request placeholder -->
<!--- ^ DO NOT REMOVE placeholder above ^ -->



### Fixed Issues (if relevant)
<!---
    If relevant, please provide a list of fixed issues in the format magento/catalog-storefront#<issue_number>.
    There could be 1 or more issues linked here and it will help us find some more information about the reasoning behind this change.
-->
1. Fixes https://github.com/magento/catalog-storefront/issues/465


### Questions or comments
<!---
	If relevant, here you can ask questions or provide comments on your pull request for the reviewer
	For example if you need assistance with writing tests or would like some feedback on one of your development ideas
-->

### Code Review Checklist (*)


- [ ] Story AC is completed
- [ ] proposed changes correspond to [Magento Technical Vision](https://devdocs.magento.com/guides/v2.2/coding-standards/technical-guidelines.html)
- [ ] new or changed code is covered with web-api/integration tests (if applicable)
- [ ] no backward incompatible changes

aleksejsbaranovs committed:
Price change notification
- Price change events

aleksejsbaranovs committed:
Price change notification
- Price change events

aleksejsbaranovs committed:
Price change notification
- Base price / special price batch processing

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Merge branch 'main' of github.com:aleksejsbaranovs10/commerce-data-export into story-443-price-notifications

aleksejsbaranovs committed:
Price change notification
- Events batch processing

Commit:
Price change notification
  - Improved custom option price notifications
  - Added/Improved custom option type price notifications

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Price change notification
- Events batch processing

Commit:
Price change notification
 -Added amqp

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Merge branch 'main' of github.com:aleksejsbaranovs10/commerce-data-export into story-443-price-notifications

Commit:
Price change notification
-Reworked message formatting for tier prices, customizable prices and normal prices

aleksejsbaranovs committed:
Price change notification
- Events batch processing

aleksejsbaranovs committed:
Price change notification
- Price events

aleksejsbaranovs committed:
Price change notification
- Price events

aleksejsbaranovs committed:
Price change notification
- Price events

aleksejsbaranovs committed:
Price change notification
- Static tests fix

aleksejsbaranovs committed:
Price change notification
    - Price events

Commit:
Sync prices
-Added product price full sync implementation with batching
-Temporarily disabled the rest of the prices providers for development

aleksejsbaranovs committed:
Price change notification
- Integration test fix

aleksejsbaranovs committed:
Merge branch 'main' of github.com:aleksejsbaranovs10/commerce-data-export into story-443-price-notifications

mslabko commented on app/code/Magento/CatalogPriceDataExporter/Model/Indexer/ProductPricesFeedIndexer.php:
```diff
@@ -0,0 +1,146 @@
+/**
+ * Class responsible for indexing product prices and provide price change events
+ */
+class ProductPricesFeedIndexer implements IndexerActionInterface, MviewActionInterface
```
It's better to create a separate model, which will collect price data instead of depending on indexer functionality. 
From indexer, we need only the "full reindex" feature.

For partial updates we may use indexer but only for the ability to use mview functionality

bricht replied:
Done

Commit:
Merge branch 'story-443-price-notifications' into 465_price-full-sync

Commit:
Prices full reindex
-Added separate providers for full reindex
-Reorganised class structure to fit requirements

bricht commented:
@magento run all tests

Commit:
Prices full reindex
-di fix

bricht commented:
@magento run all tests

Commit:
Prices full reindex
-Code cleanup

bricht commented:
@magento run all tests

Commit:
Full price sync
-Static fixes

Commit:
Full price synchronisation
-Moved full price synchronisation to cli command, in order to decouple from unnecessary general indexer logic

Commit:
Full price sync
-Modified cli command to support synchronizing by single price type and moved the logic out of the cli command itself
-Added abstract test and 2 tests
-Other minor changes

bricht commented:
@magento run all tests

bricht commented:
Unfortunately I didn't finish adding tests in time. I added Abstract test and tests for bundle and simple product prices so these are functioning examples to make the other tests.
We are still lacking tests for:
configurable_variation, 
tier_price,
downloadable_link_price,
custom_option_type_price,
custom_option_price.

mslabko commented:
@magento import pr to magento-commerce/commerce-data-export

magento-engcom-team commented:
@mslabko the pull request successfully imported.
